geojson/geojson_shapes_util.go

```diff
@@ -391,6 +391,10 @@ var GlueBytes = []byte("##")
 // can be used later while filering the doc values.
 func NewGeometryCollection(coordinates [][][][][]float64,
 	typs []string) (index.GeoJSON, []byte, error) {
+	// If the user enters typs as nil, this will prevent a panic.
+	if typs == nil || len(typs) < len(coordinates) {
+		return nil, nil, fmt.Errorf("types shorter than coordinates")
```

Comment:
Better error message here,
"types shorter than coordinates" => "missing type information for some shapes" or something better